public/app/plugins/panel/timeseries/plugins/annotations/AnnotationMarker.tsx

```diff
@@ -82,17 +83,46 @@ export function AnnotationMarker({ annotation, timeZone, style }: Props) {
     [timeZone]
   );
 
+  const canDelete = useCallback(() => {
+    let canDelete = true;
+
+    if (contextSrv.accessControlEnabled()) {
+      if (annotation.dashboardId !== 0) {
+        canDelete = !!panelCtx.annotationPermissions?.dashboard.canDelete;
```

Comment:
Wonder if this and the assignment below can be abstracted into a function, since it's used in a lot of places.

Reply:
I made a small refactor moving this logic to a function in a parent. I'm not sure we can extract a function with this since it has a lot of parameters and conditions.